pennylane_qiskit/qiskit_device2.py

```diff
@@ -581,36 +556,49 @@ def _execute_runtime_service(self, circuits, session):
     def _execute_sampler(self, circuit, session):
         """Execution for the Sampler primitive"""
 
-        qcirc = circuit_to_qiskit(circuit, self.num_wires, diagonalize=True, measure=True)
-        if circuit.shots:
-            self.options.execution.shots = circuit.shots.total_shots
-        sampler = Sampler(session=session, options=self.options)
+        qcirc = [circuit_to_qiskit(circuit, self.num_wires, diagonalize=True, measure=True)]
+        sampler = Sampler(session=session)
+        compiled_circuits = self.compile_circuits(qcirc)
 
-        result = sampler.run(qcirc).result()
-        self._current_job = result
+        result = sampler.run(compiled_circuits).result()[0]
+        attr = dir(result.data)[-1]
+        self._current_job = getattr(result.data, attr)
+
+        results = []
```

Comment:
I don't like this code and would prefer to just use `data.meas.get_counts()` as described in the [docs](https://docs.quantum.ibm.com/api/migration-guides/v2-primitives#run-multiple-experiments-in-a-single-job-1), but it seems that somehow the transpilation process renames this attribute according to what the classical register is named. Basically, `meas` is not guaranteed to be an attribute that you can access and call `get_counts()` on. So this is the solution I've arrived on for now.